src/sql/workbench/contrib/notebook/browser/notebookExplorer/notebookExplorerViewlet.ts

```diff
@@ -90,30 +124,273 @@ export class NotebookExplorerViewPaneContainer extends ViewPaneContainer {
 		@IConfigurationService configurationService: IConfigurationService,
 		@IMenuService private menuService: IMenuService,
 		@IContextKeyService private contextKeyService: IContextKeyService,
-		@IViewDescriptorService viewDescriptorService: IViewDescriptorService
+		@IViewDescriptorService viewDescriptorService: IViewDescriptorService,
+		@IFileService private readonly fileService: IFileService,
 	) {
 		super(VIEWLET_ID, { mergeViewWithContainerWhenSingleView: true }, instantiationService, configurationService, layoutService, contextMenuService, telemetryService, extensionService, themeService, storageService, contextService, viewDescriptorService);
+		this.inputBoxFocused = Constants.InputBoxFocusedKey.bindTo(this.contextKeyService);
+		this.triggerQueryDelayer = this._register(new Delayer<void>(0));
+		this.queryBuilder = this.instantiationService.createInstance(QueryBuilder);
 	}
 
 	create(parent: HTMLElement): void {
 		addClass(parent, 'notebookExplorer-viewlet');
 		this.root = parent;
 
+		this.searchWidgetsContainerElement = append(this.root, $('.header'));
+		this.createSearchWidget(this.searchWidgetsContainerElement);
+
 		this.notebookSourcesBox = append(this.root, $('.notebookSources'));
 
 		return super.create(this.notebookSourcesBox);
 	}
 
+	private createSearchWidget(container: HTMLElement): void {
+		this.searchWidget = this._register(this.instantiationService.createInstance(NotebookSearchWidget, container, <INotebookExplorerSearchOptions>{
+			value: '',
+			replaceValue: undefined,
+			isRegex: false,
+			isCaseSensitive: false,
+			isWholeWords: false,
+			searchHistory: [],
+			replaceHistory: [],
+			preserveCase: false
+		}));
+
+
+		this._register(this.searchWidget.onSearchSubmit(options => this.triggerQueryChange(options)));
+		this._register(this.searchWidget.onSearchCancel(({ focus }) => this.cancelSearch(focus)));
+		this._register(this.searchWidget.searchInput.onDidOptionChange(() => this.triggerQueryChange()));
+
+		this._register(this.searchWidget.onDidHeightChange(() => this.searchView?.reLayout(getTotalHeight(this.searchWidgetsContainerElement))));
+
+		this._register(this.searchWidget.onPreserveCaseChange((state) => {
+			if (this.searchView && this.searchView.searchViewModel) {
+				this.searchView.searchViewModel.preserveCase = state;
+			}
+			this.refreshTree();
+		}));
+
+		this._register(this.searchWidget.searchInput.onInput(() => this.searchView?.updateActions()));
+
+		this.trackInputBox(this.searchWidget.searchInputFocusTracker);
+	}
+
+	cancelSearch(focus: boolean = true): boolean {
+		if (focus) {
```

Comment:
Is it intended that this doesn't seem to be doing anything about actually cancelling the search?

Reply:
@MaddyDev Ping on this

Reply:
I updated it, thanks!

---
In reply to: [447856716](https://github.com/microsoft/azuredatastudio/pull/11117#discussion_r447856716) [](ancestors = 447856716)